packages/ai-jsx/src/react/index.ts

```diff
@@ -0,0 +1,84 @@
+import * as ReactModule from 'react';
+import * as LLMx from '../index.js';
+import { Serialize } from './serialize.js';
+export * from '../index.js';
+
+export declare namespace JSX {
+  type ElementType = ReactModule.JSX.ElementType | LLMx.JSX.ElementType | typeof React | typeof jsx;
+  type Element = ReactModule.JSX.Element & LLMx.Node;
+  type IntrinsicElements = ReactModule.JSX.IntrinsicElements;
+  type ElementChildrenAttribute = ReactModule.JSX.ElementChildrenAttribute & LLMx.JSX.ElementChildrenAttribute;
+}
+
+export function createElement(...args: Parameters<typeof ReactModule.createElement>) {
+  const tag = args[0];
+  const reactElement = ReactModule.createElement(...args);
+  const aiElement = LLMx.createElement(
+    tag === ReactModule.Fragment ? LLMx.Fragment : (tag as any),
+    args[1] as any,
+    ...args.slice(2)
+  );
+  return LLMx.makeIndirectNode(reactElement, aiElement);
+}
+
+export const Fragment = ReactModule.Fragment;
+
+export function useAI(children: LLMx.Node) {
+  const [result, setResult] = ReactModule.useState([] as ReactModule.ReactNode);
+  const [isDone, setIsDone] = ReactModule.useState(false);
+
+  ReactModule.useEffect(() => {
+    let shouldStop = false;
+    async function stream() {
+      setResult([]);
+      setIsDone(false);
+
+      // TODO: add a way for a render context to be aborted
+      const renderResult = LLMx.createRenderContext().render(children, {
+        stop: (e) => e.tag == React,
+        map: (frame) => frame,
```

Comment:
Why do we need a map if it's an identity function?

Reply:
A while back I decided that yielding the stopped elements by default was a footgun that shouldn't be the default. So the default `map` is _not_ identity, but rather "filter out elements matched by `stop` and `.join("")` the rest". Will rework a bit to make it somewhat more idiomatic. (The stopping thing is what makes things awkward which should ultimately move down into core.)